test/services/database/challenge_repository_test.dart

```diff
@@ -128,6 +128,39 @@ void main() {
       expect(challenges.length, 1);
       expect(challenges.first.postId, posts[1].id); // the other post
     });
+
+    group("Regression tests", () {
+      //It seems that if you don't have any pastChallenges the mocked firestore
+      // will don't do any check on the query, so we need to add a pastChallenge to
+      // the user.
+      test("Challenge getter when more than 30 available posts works",
+          () async {
+        //Create a post that will be used as a completed past challenge.
+        final pastPost = PostDataGenerator.generatePostData(1).first;
+        final pastPostId =
+            await postRepository.addPost(pastPost, inChallengeRange);
+
+        //Set the post as a past challenge
+        final data = ChallengeGenerator.generate();
+        await firestore
+            .collection(UserFirestore.collectionName)
+            .doc(uid.value)
+            .collection(ChallengeFirestore.pastChallengesSubCollectionName)
+            .doc(pastPostId.value)
+            .set(data.toDbData());
+
+        //Create 30 posts and get the challenges
+        FirestorePostGenerator generator = FirestorePostGenerator();
+        var challenges = await challengeRepository.getChallenges(uid, userPos);
```

Comment:
I am not sure why the use of `var` was chosen here, as `challenges` is not used before its final reassignment (using the exact same expression).

I think I would just make more sense like this:

```dart
final generator = FirestorePostGenerator();
await generator.addPostsReturnDataOnly(firestore, inChallengeRange, 30);

final challenges =
    await challengeRepository.getChallenges(uid, userPos);
```

Maybe there is a side effect that I am not aware of that would make my suggestion not behave identically ?

Reply:
You are completely right; getting two times the challenges of the user is not useful, as the first time none will be returned. Moreover it does not have any side effect that could affect the test.

Fixed in: [671e885](https://github.com/ProximaEPFL/proxima/pull/261/commits/671e8850f2640b10537e089b8443cc704b90d3ef)

Thank you 🚀